Kernel/Prekernel/init.cpp

```diff
@@ -94,6 +98,13 @@ extern "C" [[noreturn]] void init()
         auto& kernel_program_header = kernel_program_headers[i];
         if (kernel_program_header.p_type != PT_LOAD)
             continue;
+        // we require that the ELF kernel is a DROW
+        if (kernel_program_header.p_align < PAGE_SIZE)
+            halt();
+        if (kernel_program_header.p_vaddr != kernel_program_header.p_offset)
+            halt();
+        if (kernel_program_header.p_filesz != kernel_program_header.p_memsz)
+            halt();
```

Comment:
Can we combine these checks into a helper function to make the code a bit more descriptive?
E.g `is_drow_kernel()` or some such.